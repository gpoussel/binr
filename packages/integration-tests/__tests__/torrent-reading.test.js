"use strict";

const fs = require("fs");
const { BinaryReader } = require("@binr/binary-reader");
const { DefinitionReader } = require("@binr/definition-reader");

const pathToBinaryFixtures = `${__dirname}/../__fixtures__/binaries`;
const pathToDefinitionFixtures = `${__dirname}/../__fixtures__/definitions`;

describe("TORRENT reading", () => {
  const definitionReader = new DefinitionReader();

  const torrentDefinitionFile = fs.readFileSync(`${pathToDefinitionFixtures}/torrent.binr`);
  const torrentDefinition = definitionReader.readInput(torrentDefinitionFile.toString());

  const torrentFile = `${pathToBinaryFixtures}/ubuntu.torrent`;
  const torrentBuffer = fs.readFileSync(torrentFile);

  test("reads TORRENT file", () => {
    const binaryReader = new BinaryReader();
    const torrentValue = binaryReader.read(torrentBuffer, torrentDefinition, "TorrentFile");
    expect(torrentValue).toBeDefined();
    expect(torrentValue.identifier).toBe("d");
  });
});
