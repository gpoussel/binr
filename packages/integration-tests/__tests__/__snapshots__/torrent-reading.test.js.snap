// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TORRENT reading reads TORRENT file 1`] = `
Object {
  "identifier": "d",
}
`;
